### DISCUSSION TITLE
Remove generated types redundancy via field interfaces

### DISCUSSION THREAD

**Discussion by yandodov** [payloadcms/payload#5640] [Feature Requests & Ideas]
We have a group of meta fields (under a `meta` field of type `group`) that we have on all globals and on some collections, like this:

```ts
export const metaField = {
  name: "meta",
  type: "group",
  fields: [
    {
      name: "title",
      type: "text",
    },
    {
      name: "description",
      type: "textarea",
    },
    // ...
  ],
};
```

```ts
import { metaField } from "./fields/metaField";

export const Global1 = {
  slug: "global-1",
  fields: [
    metaField,
    // ...other fields
  ],
};
```

However, this results in fields having the following types:

```ts
export interface Global1 {
  id: string;
  meta: {
    title?: string | null;
    description?: string | null;
  };
  // ...
  updatedAt?: string | null;
  createdAt?: string | null;
}

export interface Global2 {
  id: string;
  meta: {
    title?: string | null;
    description?: string | null;
  };
  // ...
  updatedAt?: string | null;
  createdAt?: string | null;
}

// etc.
```

This means that every single global has a copy of the fields in `meta`, which are identical.

---

Fields could have a separate `interface` attribute that could hint Payload to generate a separate TypeScript interface for it, like so:

```diff
export const metaField = {
  name: "meta",
  type: "group",
+ interface: 'Meta',
  fields: [
    // ...
  ],
};
```

Then, the types could look like so:

```diff
+ export interface Meta {
+   title?: string | null;
+   description?: string | null;
+ }

export interface Global1 {
  id: string;
+ meta: Meta;
- meta: {
-   title?: string | null;
-   description?: string | null;
- };
  updatedAt?: string | null;
  createdAt?: string | null;
}

export interface Global2 {
  id: string;
+ meta: Meta;
- meta: {
-   title?: string | null;
-   description?: string | null;
- };
  updatedAt?: string | null;
  createdAt?: string | null;
}
```

This way, `payload-types.ts` will be shorter.

Also, this offers users to import the field type for usage in other files. Otherwise, you have to import a global type and access the types this way. In this example, you have to import either `Global1` or `Global2` and use `Global1["meta"]`, which is confusing. Importing a `Meta` interface would be much better.

**Comment by javierlinked** [comment]
Is `interfaceName` what you are looking for? It's part of `BlockField`.

**Reply by yandodov** [reply]
It appears that this exists on the group field as well and does exactly what I want. Thank you!
